polylogue/util.py

```
@@ -59,8 +59,23 @@ def parse_rfc3339_to_epoch(ts: Optional[str]) -> Optional[float]:
         return None
 
 
-def parse_input_time_to_epoch(s: Optional[str]) -> Optional[float]:
-    if not s:
+def parse_input_time_to_epoch(s: Optional[Union[str, float, int, datetime.date, datetime.datetime]]) -> Optional[float]:
+    if s is None:
+        return None
+    if isinstance(s, (int, float)):
+        try:
+            return float(s)
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)